### DISCUSSION TITLE
Copying repositories

### DISCUSSION THREAD

**Discussion by Frankenstein6851** [community/community#30963] [New to GitHub]
I am new to git and have some repositories on a machine that is about to be shutdown and rebuilt. I need to make
a copy of these so I restore them after the machine rebuild is complete. I have been strictly forbidden from putting
anything on github.com for proprietary reasons. How does one copy a repository without using github?

**Comment by airtower-luna** [comment]
Just copy the repository directory with everything in it, that's all that's needed. The critical part is the `.git` directory. That's where the history, branches, tags, and all are stored.

Keep in mind that you can clone/fetch/pull *any* Git repositories, not only ones on GitHub or any other remote host. A local directory works just as well. As [Git Basics - Working with Remotes](https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes) describes it:

> It is entirely possible that you can be working with a “remote” repository that is, in fact, on the same host you are. The word “remote” does not necessarily imply that the repository is somewhere else on the network or Internet, only that it is elsewhere. Working with such a remote repository would still involve all the standard pushing, pulling and fetching operations as with any other remote.

So you could also do something like `git clone /your/repo/path` to clone your local repository to your backup location.

If your files are big and you want to save some space you could [create a *bare* clone](https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---bare) on your backup device. A bare repository has no checked out files, so it's basically like *only* the `.git` repository. From which you can later make a regular clone again. Note that `git clone` also [has a `--mirror` option](https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---mirror), which may or may not be relevant depending on how you use the repository.

**Comment by MrMKN** [comment]
https://github.com/Decoderseeman/mkv-renamer-pyro
